### DISCUSSION TITLE
Is it possible to narrow source initially?

### DISCUSSION THREAD

**Discussion by WeissP** [minad/consult#894] [Q&A] [answered]
Is it possible to narrow a source initially?

To prevent the XY problem, here is my motivation to ask this:
I am creating a new command for finding specific files in different directories via `consult--multi`. Each source searches files in a specific directory. What I want to do is to allow users focusing on a particular source if the current buffer is located within the directory associated with the given source. For example:

- source A with dir `/a/`
- source B with dir `/b/`

If the path of the current buffer is `/a/x.txt`, I want user to focus on source A. If the path of the current buffer is `/c/x.txt`, then both sources should be displayed as usual.

So far, I have attempted to dynamically control the `:hidden` key for each source before invoking `consult--multi`. For instance, Source B will be hidden if the user is in directory `/a`. However, I have found that even if a source is hidden, the command defined in `:new` can still be invoked. For example, if the user is in directory `/a` and wants to create a new file using the function defined in Source A's `:new`, it is possible that the function defined in Source B may be invoked instead.

**Comment by minad** [comment]
> For example, if the user is in directory /a and wants to create a new file using the function defined in Source A's :new, it is possible that the function defined in Source B may be invoked instead.

Do you have a minimal recipe which demonstrates this issue?

**Reply by WeissP** [reply]
Run the following code, type characters to invoke `new` command (e.g., abc), then press enter. You will see something like `[a] new with arg: fasdf` in the minibuffer. But the source A is actually hidden.

``` lisp
;; -*- lexical-binding: t -*-

(setq a '(
          :name     "a"
          :category TEST
          :hidden t
          :new (lambda (cand) (message "[a] new with arg: %s" cand))
          :action (lambda (cand) (message "[a] action with arg: %s" cand))
          :items (lambda () '("a1" "a2"))                
          ))

(setq b '(
          :name     "b"
          :category TEST
          :hidden nil
          :new (lambda (cand) (message "[b] new with arg: %s" cand))
          :action (lambda (cand) (message "[b] action with arg: %s" cand))
          :items (lambda () '("b1" "b2"))                
          ))

(consult--multi (list a b) :prompt "test: ")

```

**Reply by minad** [reply, marked as answer]
Thanks! Fixed in https://github.com/minad/consult/commit/26e71bd8105763ba041f7bcc252a1af1fddff9d4. And congrats on the first fixed bug after the 1.0 release ;)
